File: common_static/js/components/_amplitude.js
/* Amplitude Analytics */

var __AMPLITUDE;
;(function() {
    var _this   = '';

    __AMPLITUDE = {
        init: function() {
            _this = this;
        },

        event: function(name) {
            events = {
                'mortgage_deal_created': 'mortgage deal created',
                'mortgage_deal_won': 'mortgage deal won',
                'mortgage_quote_created': 'mortgage quote created',
                'mortgage_quote_email_sent': 'mortgage quote email sent',
                'mortgage_product_quoted': 'mortgage product quoted',

                'motor_deal_created': 'motor deal created',
                'motor_deal_won': 'motor deal won',
                'motor_quote_created': 'motor quote created',
                'motor_quote_email_sent': 'motor quote email sent',
                'motor_product_quoted': 'motor product quoted',
                'pdf_quote_downloaded': 'pdf quote downloaded',
                'motor_order_created': 'motor order created',
                'motor_policy_saved': 'motor policy saved',
                'policy_text_extracted': 'policy text extracted',
                'motor_policy_renewed': 'motor policy renewed',

                'vehicle_valuation_checked': 'vehicle valuation checked',

                'document_previewed': 'documents previewed',
                'document_renamed': 'documents renamed'
            };

            if(name in events)
                return events[name];

            return name;
        },

        setUserData: function() {
            if(typeof amplitude != 'object')
                return;

            amplitude.getInstance().setUserId(loggedin_user_data.ID);

            var userProperties = {
                email: loggedin_user_data.EMAIL,
                company: loggedin_user_data.COMPANY
            };

            amplitude.getInstance().setUserProperties(userProperties);
        },

        logEvent: function(event_type, properties) {
            if(typeof amplitude != 'object')
                return;

            _this.setUserData();

            if(properties === undefined) properties = {};

            properties['company_id'] = current_company_info.ID;
            properties['company_name'] = current_company_info.NAME;

            amplitude.getInstance().logEvent(event_type, properties);
        }
    };

    jQuery(function() {
        __AMPLITUDE.init();
    });

})();

File: core/amplitude.py
"""Amplitude events"""
import json
import uuid

import requests
from django.conf import settings


class Amplitude:
    EVENTS = {
        'motor_deal_created': 'motor deal created',
        'motor_deal_won': 'motor deal won',
        'motor_quote_created': 'motor quote created',
        'motor_quote_email_sent': 'motor quote email sent',
        'motor_product_quoted': 'motor product quoted',
        'pdf_quote_downloaded': 'pdf quote downloaded',
        'motor_order_created': 'motor order created',
        'motor_policy_saved': 'motor policy saved',
        'policy_text_extracted': 'policy text extracted'
    }

    def __init__(self, request):
        self.api_url = 'https://api.amplitude.com/httpapi'
        self.api_key = settings.AMPLITUDE['API_KEY']
        self.request = request

    def log_event(self, event_type='', properties={}):
        properties['company_id'] = self.request.company.pk
        properties['company_name'] = self.request.company.name

        event = {
            'user_id': uuid.uuid4().hex,
            'device_id': None,
            'event_type': event_type,
            'event_properties': properties
        }

        if self.request.user.pk:
            event['user_id'] = self.request.user.pk

        response = requests.post(
            self.api_url,
            data=[
                ('api_key', self.api_key),
                ('event', json.dumps([event])),
            ],
            timeout=5)

        return response
